app/diversity/post.controller.js

```diff
@@ -0,0 +1,45 @@
+const { body } = require('express-validator')
+const { setSentencePlanComment } = require('../../common/data/sentencePlanningApi')
+const { getDiversity } = require('./get.controller')
+const { countWords } = require('../../common/utils/util')
+
+const wordsAllowed = 250
+let tooManyWords = false
+
+const validationRules = () => {
+  return [
+    body('diversity')
+      .isLength({ min: 1 })
+      .withMessage('Record how you will take account of diversity factors'),
+    body('diversity')
+      .custom(value => {
+        tooManyWords = countWords(value) > wordsAllowed
+        return !tooManyWords
+      })
+      .withMessage('Response to diversity factors must be 250 words or fewer'),
+  ]
+}
+
+const postDiversity = async (req, res) => {
+  if (req.errors) {
+    const renderInfo = {}
+    let wordsOver = false
+    if (tooManyWords || req.tooManyWords) {
+      wordsOver = countWords(req.body.diversity) - wordsAllowed
+    }
+    renderInfo.wordsOver = wordsOver
+    req.renderInfo = renderInfo
```

Comment:
I really don't like this; we're effectively treating tooManyWords as a global.
Can we not just scrap the `tooManyWords` var and calculate words over and return if its greater than 0?

wordsOver = countWords(req.body.diversity) - wordsAllowed
req.renderInfo = { wordsOver: wordsOver > 0 ? wordsOver :  0}

Reply:
I did have a bad feeling about this one - will look at your suggestion and revisit